cobc/parser.y

```
@@ -18134,13 +18134,40 @@ on_size_error_phrases:
   %prec SHIFT_PREFER
   {
 	/* no [NOT] ON SIZE ERROR is specified (= no explicit handling) */
+	if (CB_EXCEPTION_ENABLE(COB_EC_SIZE)
+		|| CB_EXCEPTION_ENABLE(COB_EC_SIZE_ADDRESS)
+		|| CB_EXCEPTION_ENABLE(COB_EC_SIZE_EXPONENTIATION)
+		|| CB_EXCEPTION_ENABLE(COB_EC_SIZE_IMP)
+		|| CB_EXCEPTION_ENABLE(COB_EC_SIZE_OVERFLOW)
+		|| CB_EXCEPTION_ENABLE(COB_EC_SIZE_TRUNCATION)
+		|| CB_EXCEPTION_ENABLE(COB_EC_SIZE_UNDERFLOW)
+		|| CB_EXCEPTION_ENABLE(COB_EC_SIZE_ZERO_DIVIDE))
+	{
+		current_statement->handler_type = SIZE_ERROR_HANDLER;
+		current_statement->ex_handler = NULL;
+	}
   }
 | on_size_error _not_on_size_error
 | not_on_size_error _on_size_error
   {
 	if ($2) {
 		cb_verify (cb_not_exception_before_exception,
 			_("NOT SIZE ERROR before SIZE ERROR"));
+	} else if ($1 && $1 != cb_int1) {
```

$1 here is always set as `not_on_size_error` is mandatory - you can drop the check for this and the setting to cb_int2 below

Nice and correct addition for that "null handler" in the parser BTW!


... but: it should not be added here by the standard (I think also by previous standards).
To quote: 

> If the EC-SIZE exception condition was raised and a SIZE ERROR phrase is not specified, processing
proceeds as specified in 14.6.13.1.3, Fatal exception conditions. If a NOT SIZE ERROR phrase is specified,
it is ignored.

The point here if one explicit codes `NOT SIZE ERROR` than this is identical to a coded `ON SIZE ERROR CONTINUE`

---

I think this phrase means that if the exceptions exists and the `NOT SIZE ERROR` phrase is specified, then the `NOT SIZE ERROR` phrase is ignored, not that the exception is ignored

---

I'm sure this isn't the case. You may try some computers, but the ones I've tested behave like my understanding of the standard. 
Logically: coding only NOT ON SIZE ERROR means you kind of expect but ignore one and execute the "important" part only if there isn't an error. 
Also note that coding it means (per standard) that the target field is unchanged if an error happens, otherwise it is undefined if no checking is on.

---

Sure but the phrasing is `If a NOT SIZE ERROR phrase is specified, it is ignored.`, not `If a NOT SIZE ERROR phrase is specified, the exception is ignored.` so `it` must refer to the subject of the phrase, not something that was in a phrase before.

In addition this phrase appears after the description of what should happen in case of an exception without an `ON SIZE ERROR` phrase, and after specifically saying `and processing proceeds as specified in ...` so logically every steps the standard specify should happen before any consideration for `NOT ON SIZE ERROR`.

Afaik not many (if any) implementers have an exception system in their compiler, so maybe they work differently when having `NOT ON SIZE ERROR`, but if we decide to implement one, we should adhere to the standard as close as possible.

---

let's take the following approach: we surround that with `#if 0 /* CHECKME: possible dialect configuration */` and get the code upstream, then check what implementations do and also with the standard committee what the intention is